File: generated-sources/php-ze-ph/mojang-authentication/src/App/com/github/asyncmc/mojang/authentication/php/ze/ph/model/Agent.php
<?php

namespace com.github.asyncmc.mojang.authentication.php.ze.ph.server\com.github.asyncmc.mojang.authentication.php.ze.ph.model;

use Articus\DataTransfer\Annotation as DTA;

/**
 * Identifies the software doing the request
 */
class Agent
{
    /**
     * The name of the game
     * @DTA\Data(field="name", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $name;
    /**
     * The agent version, usually 1.
     * @DTA\Data(field="version", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"int"})
     * @var int
     */
    public $version;
}

File: generated-sources/kotlin-spring/mojang-authentication/src/main/kotlin/com/github/asyncmc/mojang/authentication/kotlin/spring/model/Agent.kt
package com.github.asyncmc.mojang.authentication.kotlin.spring.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonValue
import javax.validation.Valid
import javax.validation.constraints.*

/**
 * Identifies the software doing the request
 * @param name The name of the game
 * @param version The agent version, usually 1.
 */
data class Agent (

        @JsonProperty("name") val name: Agent.Name? = null,

        @JsonProperty("version") val version: Int? = null
) {

    /**
    * The name of the game
    * Values: MINECRAFT,SCROLLS
    */
    enum class Name(val value: String) {
    
        @JsonProperty("Minecraft") MINECRAFT("Minecraft"),
    
        @JsonProperty("Scrolls") SCROLLS("Scrolls");
    
    }

}

